### PR TITLE
Handle notes, errors and warnings

### PR DESCRIPTION
**Pull request by koddsson**
This PR moves more functionality from `github/github` into this element. Since we don't need to do HTML parsing/validation any more, this element can handle more responsibilities.

We handle the showing and hiding of note/error/warning elements in `<auto-check>`. This simplifies the relationship between the element and the application code.

We also not add the whole response into the success and failure events that we dispatch.

### CODE REVIEW

**Comment by koddsson** [comment]
Need to update the README

File: src/index.js [hunk 1]
```diff
@@ -43,6 +43,18 @@ export default class AutoCheckElement extends HTMLElement {
...
+  get note(): HTMLElement {
+    let note = this.querySelector(this.noteSelector)
+    if (!note) {
+      note = document.createElement('p')
+    }
+    return note
+  }
+
+  get noteSelector(): string {
+    return this.getAttribute('data-note-selector') || 'p, .error, .warning'
```
Comment (muan) [inline src/index.js]: I don't think this should fallback to such generic selectors. Setting `innerHTML` should be very intentional. In `<details-menu>` we use `data-menu-button-text` for `textContent` and `data-menu-button-contents` for `innerHTML`.

Comment (koddsson) [reply src/index.js]: Yeah I was trying to be a bit cheeky so I didn't have to update all the instances in `github/github` 😄

Comment (koddsson) [reply src/index.js]: And it was a bit simpler then having a note and then an error/warning element. I'm gonna tackle it and see if I can make them separate but still have the API simple.

**Comment by koddsson** [comment]
> The custom element is responsible for managing the network request lifecyle of the validation request and updating the input's validity state. The host app is responsible for managing any user interface changes and styles based on the success or error state of the validation. Separating behavior from presentation decouples the element from any particular host application using it.

I think I'd like to talk about the "The host app is responsible for managing any user interface changes" part.

Is there any reason why this should be handled by the host app and not the element? If it's handled by the element, the host app doesn't need to do anything additionally but style it however it wants. When the host app handles the UI changes, it needs to handle the data and set the custom validity based on the data it receives.

---

The way I wanted to go was making the element handle more and users would be able to drop it into their application, set the correct server response and style it and they'd be up and running.


A instance where the element will not show any message and leaves that up to the host app during event delegation:
```js
<auto-check>
  <input />
</auto-check>
```

Adding a "message" element will make the `<auto-check>` element toggle the error message for you without any extra code from the host application:
```js
<auto-check>
  <input />
  <p data-message-element></p>
</auto-check>
```

**Pull request closed without merging**